src/app/pages/shopping-list/shopping-edit/shopping-edit.component.html

```diff
@@ -1,21 +1,21 @@
 <div class="row">
   <div class="form">
-    <form>
+    <form (ngSubmit)="onSubmit(f)" #f="ngForm">
       <div class="form-row">
         <div class="form-group-1 form-group">
           <label for="name">Name</label>
-          <input type="text" id="name" class="form-controle" #nameInput>
+          <input type="text" id="name" class="form-controle" name="name" ngModel required>
```

Comment:
It's ok to use both ngModel, while learning, but in future it will be better to choose formControls or ngModels

Reply:
ok